subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/ErrorHandlingModuleComponentRepository.java

```diff
@@ -94,15 +102,30 @@ public ModuleComponentRepositoryAccess getRemoteAccess() {
     }
 
     private static final class ErrorHandlingModuleComponentRepositoryAccess implements ModuleComponentRepositoryAccess {
+        private static final Logger LOGGER = Logging.getLogger(ErrorHandlingModuleComponentRepositoryAccess.class);
+        private final static String MAX_TENTATIVES_BEFORE_BLACKLISTING = "org.gradle.internal.repository.max.tentatives";
```

Comment:
I'd be in favor of making those system properties into proper build options, as everyone's environment is different. E.g. in China you might need very generous timeouts while in Germany you might prefer faster failures.